ckanext/xloader/utils.py

```diff
@@ -9,6 +9,36 @@
 from decimal import Decimal
 
 import ckan.plugins as p
+from ckan.plugins.toolkit import config
+
+# resource.formats accepted by ckanext-xloader. Must be lowercase here.
+DEFAULT_FORMATS = [
+    "csv",
+    "application/csv",
+    "xls",
+    "xlsx",
+    "tsv",
+    "application/vnd.ms-excel",
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "ods",
+    "application/vnd.oasis.opendocument.spreadsheet",
+]
+
+
+class XLoaderFormats(object):
+    formats = None
+
+    @classmethod
+    def is_it_an_xloader_format(cls, format_):
+        if cls.formats is None:
+            cls._formats = config.get("ckanext.xloader.formats")
+            if cls._formats is not None:
```

Comment:
I get that it's important to use `is not None` rather than just testing truthiness, because if the config specifies an empty list, we should use an empty list. Still, this results in a double negative, so perhaps it would be better structured as "if is None, then use default, else use config value".

Not actually part of this change, though.